components/styled/icon/src/molecules/icon.tsx

```diff
@@ -0,0 +1,201 @@
+    }
+
+    case 'circle': {
+      return <CircleIcon size={size} status={status ?? 'default'} {...rest} />
+    }
+
+    case 'column': {
+      return <ColumnIcon size={size} status={status ?? 'default'} {...rest} />
+    }
+
+    case 'comment': {
+      return <CommentIcon size={size} {...rest} />
+    }
+
+    case 'counter': {
+      return <CounterIcon size={size} status={status ?? 'default'} value={counterValue ?? 0} {...rest} />
+    }
+
+    case 'cross': {
+      return <CrossIcon size={size} {...rest} />
+    }
+
+    case 'edit': {
+      return <EditIcon size={size} {...rest} />
+    }
+
+    case 'ellipsis-horizontal': {
+      return <EllipsisHorizontalIcon size={size} {...rest} />
+    }
+
+    case 'ellipsis-vertical': {
+      return <EllipsisVerticalIcon size={size} {...rest} />
+    }
+
+    case 'exclamation': {
+      return <ExclamationIcon size={size} status={status ?? 'default'} {...rest} />
+    }
+
+    case 'external-link': {
+      return <ExternalLinkIcon size={size} {...rest} />
+    }
+
+    case 'filter': {
+      return <FilterIcon size={size} {...rest} />
+    }
+
+    case 'folder-plus': {
+      return <FolderPlusIcon size={size} {...rest} />
+    }
+
+    case 'grid-layout': {
+      return <GridLayoutIcon size={size} {...rest} />
+    }
+
+    case 'hamburger': {
+      return <HamburgerIcon size={size} {...rest} />
+    }
+
+    case 'info-circle': {
+      return <InfoCircleIcon size={size} status={status ?? 'default'} {...rest} />
+    }
+
+    case 'layer-group': {
+      return <LayerGroupIcon size={size} {...rest} />
+    }
+
+    case 'list-layout': {
+      return <ListLayoutIcon size={size} {...rest} />
+    }
+
+    case 'pause': {
+      return <PauseIcon size={size} {...rest} />
+    }
+
+    case 'play': {
+      return <PlayIcon size={size} {...rest} />
+    }
+
+    case 'plus': {
+      return <PlusIcon size={size} {...rest} />
+    }
+
+    case 'spinner': {
+      return <SpinnerIcon size={size} {...rest} />
+    }
+
+    case 'stop': {
+      return <StopIcon size={size} {...rest} />
+    }
+
+    case 'table': {
+      return <TableIcon size={size} status={status ?? 'default'} {...rest} />
+    }
+
+    case 'toggle-off': {
+      return <ToggleOffIcon size={size} {...rest} />
+    }
+
+    case 'toggle-on': {
+      return <ToggleOnIcon size={size} {...rest} />
+    }
+
+    case 'trash': {
+      return <TrashIcon size={size} {...rest} />
+    }
+
+    case 'user': {
+      return <UserIcon size={size} {...rest} />
+    }
+
+    case 'dot-circle': {
+      return <DotCircleIcon size={size} status={status ?? 'default'} {...rest} />
+    }
+
+    case 'search': {
+      return <SearchIcon size={size} {...rest} />
+    }
+
+    case 'checkbox': {
+      return <CheckboxIcon size={size} {...rest} />
+    }
+
+    case 'square': {
+      return <SquareIcon size={size} {...rest} />
+    }
+
+    default: {
+      return <SquareIcon size={size} {...rest} />
```

Comment:
Any reason for square icon?

Reply:
That one of Saqib's additions @saqib-airelogic so he'd be best to explain - I think it's being used as an unchecked checkbox atm

Reply:
yes its just and empty square for unchecked icon

Reply:
Ah ok, I was wondering if it'd be better to throw or have some wierd UTF uknown character than default to one of the existing icons, but the square box looks OK for this purpose. Cheers :)

Reply:
Oh right, sorry I didn't realise you were asking why it was the default not why it exists! 
To be honest it's on my TODO to come back to - Typescript is happy with the case statement without the default because the type is non-nullable, but es-lint demands one. I originally had it throw but JS then "helpfully" greys out the code as unreachable. So I went for a default but I don't like it!